From e41229022ecd2f40d5617df79949e369d219a9aa Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 3 Sep 2014 15:15:56 -0400
Subject: [ethernet] bna: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1409757361-26781-9-git-send-email-ivecera@redhat.com>
Patchwork-id: 91167
O-Subject: [RHEL7.1 PATCH 08/13] net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1136912
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ: #1136912

Upstream commit(s):
commit 537fae0101c8853abb52136788173dde74b9d1e9
Author: Jiri Pirko <jiri@resnulli.us>
Date:   Fri Jun 6 14:17:00 2014 +0200

    net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

    Signed-off-by: Jiri Pirko <jiri@resnulli.us>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/brocade/bna/bnad_ethtool.c b/drivers/net/ethernet/brocade/bna/bnad_ethtool.c
index adca62b..882cad7 100644
--- a/drivers/net/ethernet/brocade/bna/bnad_ethtool.c
+++ b/drivers/net/ethernet/brocade/bna/bnad_ethtool.c
@@ -266,8 +266,8 @@ bnad_get_settings(struct net_device *netdev, struct ethtool_cmd *cmd)
   ethtool_cmd_speed_set(cmd, SPEED_10000);
   cmd->duplex = DUPLEX_FULL;
  } else {
-  ethtool_cmd_speed_set(cmd, -1);
-  cmd->duplex = -1;
+  ethtool_cmd_speed_set(cmd, SPEED_UNKNOWN);
+  cmd->duplex = DUPLEX_UNKNOWN;
  }
  cmd->transceiver = XCVR_EXTERNAL;
  cmd->maxtxpkt = 0;
-- 
1.7.1